clap_lex/src/lib.rs

```diff
@@ -300,10 +300,14 @@ impl<'s> ParsedArg<'s> {
         self.inner == "--"
     }
 
-    /// Does the argument look like a number
-    pub fn is_number(&self) -> bool {
+    /// Does the argument look like a negative number?
+    ///
+    /// This won't parse the number in full but attempts to see if this looks
+    /// like something along the lines of `-3`, `-0.3`, or `-33.03`
+    pub fn looks_like_negative_number(&self) -> bool {
```

Comment:
I think I'd be fine with `is_negative_number`.  `looks_like_negative_number` feels a little awkward.